airflow/providers/google/cloud/triggers/cloud_run.py

```diff
@@ -0,0 +1,145 @@
+        project_id: str | None,
+        location: str = DEFAULT_BATCH_LOCATION,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: str | Sequence[str] | None = None,
+        polling_period_seconds: float = 10,
+        timeout: float | None = None,
+    ):
+        super().__init__()
+        self.project_id = project_id
+        self.job_name = job_name
+        self.operation_name = operation_name
+        self.location = location
+        self.gcp_conn_id = gcp_conn_id
+        self.polling_period_seconds = polling_period_seconds
+        self.timeout = timeout
+        self.impersonation_chain = impersonation_chain
+
+    def serialize(self) -> tuple[str, dict[str, Any]]:
+        """Serializes class arguments and classpath."""
+        return (
+            "airflow.providers.google.cloud.triggers.cloud_run.CloudRunJobFinishedTrigger",
+            {
+                "project_id": self.project_id,
+                "operation_name": self.operation_name,
+                "job_name": self.job_name,
+                "location": self.location,
+                "gcp_conn_id": self.gcp_conn_id,
+                "polling_period_seconds": self.polling_period_seconds,
+                "timeout": self.timeout,
+                "impersonation_chain": self.impersonation_chain,
+            },
+        )
+
+    async def run(self) -> AsyncIterator[TriggerEvent]:
+        """
+        Main loop of the class in where it is fetching the operation status and yields an
+        Event when done.
+
+        """
+        timeout = self.timeout
+        hook = self._get_async_hook()
+        while timeout is None or timeout > 0:
+            operation: operations_pb2.Operation = await hook.get_operation(self.operation_name)
+            if operation.done:
+                # An operation can only have one of those two combinations: if it is succeeded, then
+                # the response field will be populated, else, then the error field will be.
+                if operation.response is not None:
+                    yield TriggerEvent(
+                        {
+                            "status": RunJobStatus.SUCCESS,
+                            "job_name": self.job_name,
+                        }
+                    )
+                else:
+                    yield TriggerEvent(
+                        {
+                            "status": RunJobStatus.FAIL,
+                            "operation_error_code": operation.error.code,
+                            "operation_error_message": operation.error.message,
+                            "job_name": self.job_name,
+                        }
+                    )
+            elif operation.error.message:
+                raise AirflowException(f"Cloud Run Job error: {operation.error.message}")
+
+            if timeout is not None:
+                timeout -= self.polling_period_seconds
+
+            await asyncio.sleep(self.polling_period_seconds)
```

Comment:
On the last iteration of this while loop we call sleep only to come back to exit the loop. I think it's better to only sleep if `timeout > 0`